### PR TITLE
Initial setup and contracts

### PR DESCRIPTION
**Pull request by nini-faroux**
So this adds the initial setup and initial versions of the workflow - it includes the first couple of Plutip tests for the workflows, for `createIndex` and `createConfig`.

### CODE REVIEW

File: src/Dao/Workflow/CancelVote.purs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  , scriptCurrencySymbol
+  )
+import Contract.Value (singleton) as Value
+import Dao.Utils.Address (addressToPaymentPubKeyHash)
+import Dao.Utils.Datum (getInlineDatumFromTxOutWithRefScript)
+import Dao.Utils.Query (findUtxoByValue)
+import Data.Map as Map
+import Data.Maybe (Maybe(Nothing))
+import Data.Newtype (unwrap)
+-- import ScriptArguments.Types
```
Comment (Randy1Burrell) [inline src/Dao/Workflow/CancelVote.purs]: Commented out code should be removed

File: src/Dao/Workflow/VoteOnProposal.purs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  , TokenName
+  , Value
+  , scriptCurrencySymbol
+  )
+import Contract.Value (singleton) as Value
+import Dao.Utils.Query (findUtxoByValue)
+import Dao.Utils.Time (mkOnchainTimeRange, mkValidityRange, oneMinute)
+import Data.Maybe (Maybe(Nothing))
+import JS.BigInt (fromInt)
+-- import ScriptArguments.Types
```
Comment (Randy1Burrell) [inline src/Dao/Workflow/VoteOnProposal.purs]: Remove commented out code

File: types/LambdaBuffers/ApplicationTypes/Configuration.purs [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+  , fungibleVotePercent :: BigInt
+  }
+
+derive instance Data.Newtype.Newtype DynamicConfigDatum _
+derive instance Data.Generic.Rep.Generic DynamicConfigDatum _
+instance Data.Show.Show DynamicConfigDatum where
+  show x = Data.Show.Generic.genericShow x
+
+instance Prelude.Eq DynamicConfigDatum where
+  eq =
```
Comment (Randy1Burrell) [inline types/LambdaBuffers/ApplicationTypes/Configuration.purs:49]: Was this generated?

File: types/LambdaBuffers/ApplicationTypes/Configuration.purs [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+              ((Data.Newtype.unwrap x0).voteFungibleTokenName)
+          , Ctl.Internal.ToData.toData
+              ((Data.Newtype.unwrap x0).fungibleVotePercent)
+          ]
+        )
+    )
+
+instance Ctl.Internal.FromData.FromData DynamicConfigDatum where
+  fromData =
+    ( \x0 -> LambdaBuffers.Runtime.Plutus.casePlutusData
```
Comment (Randy1Burrell) [inline types/LambdaBuffers/ApplicationTypes/Configuration.purs:307]: Was this generated?

**Review by Randy1Burrell — APPROVED** [review]
Great work, I Ieft a few comments.

File: src/Dao/Workflow/CancelVote.purs (hunk 1 above)
Comment (nini-faroux) [reply src/Dao/Workflow/CancelVote.purs]: Good spot, will do

File: types/LambdaBuffers/ApplicationTypes/Configuration.purs (hunk 4 above)
Comment (nini-faroux) [reply types/LambdaBuffers/ApplicationTypes/Configuration.purs:307]: Exactly, these instances are generated with LambdaBuffers

**Comment by nini-faroux** [comment]
Thanks @Randy1Burrell, I'll make those changes over the weekend

**Comment by nini-faroux** [comment]
Okay @Randy1Burrell - I removed those commented out parts, going to merge this now and bring the next one up to date

**Pull request merged**